listener.go

```diff
@@ -60,17 +62,18 @@ func processSnapshot(snapshot *common.Snapshot) {
 		log.Panicf("Error starting transaction: %v", err)
 	}
 
+	// ensure that no new database updates are made on old database
```

Comment:
Well, it depends... if the plugin has threads writing to the database concurrent with the snapshot, yes. If the plugin doesn't update the db, it wouldn't matter. For example, as far as I know, VerifyAPIKey shouldn't care.